src/App.tsx

```diff
@@ -1,24 +1,97 @@
 import React from 'react';
+// eslint-disable-next-line
+import { Clock } from './components/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  nameId: number;
 };
 
-export default App;
+export const names = [
+  'Christmas Clock',
+  'Ho Ho Ho',
+  'Happy New Year',
+  'Merry Christmas',
+  'Christmas is in the Air',
+  'O Christmas Tree',
+  'Good luck',
+  'Christmas is Coming',
+  'Ding dong',
+  'Santa’s Little Helper',
+  'Jingle All the Way',
+];
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    nameId: 0,
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  changeName = () => {
+    const randomizer = Math.ceil(Math.random() * 10);
+
+    this.setState({ nameId: randomizer });
+  };
+
+  render() {
+    const { isClockVisible, nameId } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="app__title">
+          {
+            nameId
+              ? names[nameId]
+              : 'Christmas Clock'
```

Comment:
It is set by default on 0, so why don't just use `0` index?
```suggestion
            names[nameId]
```